src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -118,7 +121,10 @@
           AfterTargets="CheckTestCategories"
           Inputs="@(RunTestsForProjectInputs)"
           Outputs="$(TestsSuccessfulSemaphore);$(TestPath)%(TestTargetFramework.Folder)/$(XunitResultsFileName);$(CoverageOutputFilePath)"
-          Condition="'$(TestDisabled)'!='true'">
+          >
+    <GetTargetMachineInfo>
+      <Output TaskParameter="TargetOS" PropertyName="TargetOS" />
```

Comment:
I'm actually curious why you needed to call it again explicitly here? I would have expected TargetOS to already be set.